### PR TITLE
Add use() method to short-circuit requests with no route handlers

### PR DESCRIPTION
**Pull request by mrutkows**
It appears that the code was attempting to use the Node Express "use()" middleware to intercept and return a 500 error if the runtime was hit by any requests not ```/init``` or ```/run``` as follows:
```
app.use(function (err, req, res, next) {  // <== use without “err” parm.
    console.error(err.stack); // <== REMOVE since "err" would no longer be a valid parm.
    res.status(500).json({error: "Bad request."});
});
```

however, the method only has 3 valid parms. so it appears Node Express ignored the middle filter registration completely and the code was never invoked since it had too many parms. That is, the "err" parm. (as the first parm.) was "extra" and the signature had too many args.

This PR removes the "err" parm., as well as the now invalid console.error() call.

Please note (in addition) I fixed a typo, as well as removed an used variable (i.e., "var self") which is never used.

### CODE REVIEW

**Comment by mrutkows** [comment]
I have tested this runtime standalone and now we indeed get "500 Bad request.", whereas before we got nothing.

File: core/nodejsActionBase/app.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ app.use(bodyParser.json({ limit: "48mb" }));
 app.post('/init', wrapEndpoint(service.initCode));
 app.post('/run',  wrapEndpoint(service.runCode));
 
-app.use(function(err, req, res, next) {
+app.use(function(req, res, next) {
     console.error(err.stack);
```
Comment (rabbah) [inline core/nodejsActionBase/app.js]: is err now undefined on line 42? 
did this middleware interface change?

Comment (rabbah) [reply core/nodejsActionBase/app.js]: i see in your description you intended to remove line 42 perhaps?

Comment (mrutkows) [reply core/nodejsActionBase/app.js]: Could not find docs that supported that interface for 4.x 
https://expressjs.com/en/guide/using-middleware.html

Comment (mrutkows) [reply core/nodejsActionBase/app.js]: even in deprecated 3.x: it was not in the framework docs:
https://expressjs.com/en/3x/api.html#middleware

Comment (mrutkows) [reply core/nodejsActionBase/app.js]: I did find what likely someone copied verbatim for creating your own error handler here:https://expressjs.com/en/guide/error-handling.html
but, in standalone testing the function was never called...

Comment (mrutkows) [reply core/nodejsActionBase/app.js]: Anyone able to point to any tests that exhibit the existing impl.?

Comment (rabbah) [reply core/nodejsActionBase/app.js]: i think it's ok to remove this.

Comment (mrutkows) [reply core/nodejsActionBase/app.js]: If I hit the existing code with an POST/GET (etc.) to a bogus endpoint, I get a default 404 error:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Error</title>
</head>
<body>
<pre>Cannot POST /bad</pre>
</body>
</html>


Response code: 404 (Not Found); Time: 68ms; Content length: 143 bytes
```

if the intent is to immediately block all these bad requests immediately this change results in
```
HTTP/1.1 500 Internal Server Error
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 24
ETag: W/"18-928jOLijreS3ro9mYfVt0IU4k/0"
Date: Tue, 26 Feb 2019 20:03:54 GMT
Connection: keep-alive

{
  "error": "Bad request."
}

Response code: 500 (Internal Server Error); Time: 11928ms; Content length: 24 bytes
```

Trying to figure out the original intent and perhaps short circuit any endpoint that is not intended to be part of the actual runtime.

**Comment by mrutkows** [comment]
SO I have put back the default error handler which will only be invoked when invalid JSON is received by any of the endpoints (i.e., handle a JSON parser error) despite the fact that the default NodeJS handler catches it and properly returns ```Response code: 400 (Bad Request)```.

Why?
1) because the default handler passes back HTML in the body of the request, for example:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Error</title>
</head>
<body>
<pre>SyntaxError: Unexpected token } in JSON at position 161 ...</pre>
</body>
</html>
```

Leaving our own handler, in retrospect, actually allows us to consistently use a status code (500) we choose, as well as return JSON in the body of the response.

2) Acknowledging that receiving BAD JSON is unlikely when running under the OpenWhisk platform (Controller/Invoker); however, as we move to run under other platforms like Knative where other services may be hitting our runtimes directly (no controller) this edge case becomes much more likely.

---
BUT, I also added another use() method that actually short-circuits any requests received by NodeJS express that attempt to access routes which we have no registered handlers for.

This too is desirable going forward as we run on other platforms since we will both be registering different routes/handlers and allowing other services (or some other control plane) to hit our runtime at these different endpoints.  If indeed, we allow variant builds of our runtimes, it may become more likely situations arise where invalid routes (mismatch bw/ control plane and runtime) may occur.

**Comment by mrutkows** [comment]
Updated title to reflect my (lengthy) comments above...

**Comment by mrutkows** [comment]
@rabbah revised with current thinking captured... ready for and would appreciate your review (again)

**Pull request merged**